fix(OverflowMenu): fix border-radius and add larger size options

tjmabey opened the pull request:
This addresses https://github.com/InsideSalesOfficial/insidesales-components/issues/225

**Changes:**
* fix corners not being rounded
* adjust border-radius to 3px by request from Kellie
* add medium and large props for larger menu options
* add key to SubmenuOptionsWrapper component

**before:**
<img width="222" alt="Screen Shot 2019-03-27 at 5 23 42 PM" src="https://user-images.githubusercontent.com/18040782/55118833-52cd4e00-50b5-11e9-930d-d19028ed630f.png">

**after:**
default
<img width="224" alt="Screen Shot 2019-03-27 at 5 10 48 PM" src="https://user-images.githubusercontent.com/18040782/55118841-5b258900-50b5-11e9-8c56-c18eb84cfc39.png">
medium
<img width="223" alt="Screen Shot 2019-03-27 at 5 11 17 PM" src="https://user-images.githubusercontent.com/18040782/55118848-5f51a680-50b5-11e9-8ba4-66c4ff26eed8.png">
large
<img width="223" alt="Screen Shot 2019-03-27 at 5 11 44 PM" src="https://user-images.githubusercontent.com/18040782/55118858-64aef100-50b5-11e9-9ae5-a064dad6d93b.png">

aar0nwils0n commented:
They currently resize dynamically based on the content. Is there an issue with this approach or special use case for having fixed sizing?

tjmabey commented:
I recently chatted with Kellie Shelton about this. She gave us an updated design that has an overflowmenu at 160px width. Based on past commit history (https://github.com/InsideSalesOfficial/insidesales-components/commit/bc73d61f80848da105f453946d59a6c9fc39474d), I thought it would be better to just make several different sizes available in case design changes their preference again

aar0nwils0n commented on src/components/OverflowMenu/OverflowMenu.js:
```diff
@@ -21,7 +21,7 @@ const SelectOption = styled.div`
   box-sizing: border-box;
   display: flex;
-  padding: 0 10px;
+  padding: 0 24px 0 10px;
```
I'm curious why the right side needs different padding. If this is the case we would probably need to adjust for which direction the menu is facing

tjmabey replied:
you beat me to commenting... kellie asked for it but i see what you mean now about direction..

tjmabey replied:
i dont actually think the direction matters for this. kellie asked for 24px padding on the right edge. what would you like me to do?

aar0nwils0n replied:
Ah, I suppose text text is always right aligned anyway so it doesn't matter. It's fine as is

tjmabey commented:
Updated PR:
- removed larger menu size options
- the min-width will continue to be 120px
- added a little padding to right edge of menu options

this is from Kellie:
> minimum width will be 120 and then everything else will dynamically resize. So there will be 24px of padding from the right edge of the longest word in an overflow menu

tjmabey commented:
requesting additional review @t0v1n @richardbolt @austinknight @nathguen @gabe-lei @cozzbp @thundercat1 @jeffhoffman1 @rsmorley
